curvefs/src/mds/fs_manager.cpp

```diff
@@ -308,10 +308,17 @@ FSStatusCode FsManager::CreateFs(const ::curvefs::mds::CreateFsRequest* request,
         option_.s3AdapterOption.s3Address = s3Info.endpoint();
         option_.s3AdapterOption.bucketName = s3Info.bucketname();
         s3Adapter_->Reinit(option_.s3AdapterOption);
-        if (!s3Adapter_->BucketExist()) {
-            LOG(ERROR) << "CreateFs " << fsName
-                       << " error, s3info is not available!";
+        const Aws::String aws_key(fsName.c_str(), fsName.size());
+        if (s3Adapter_->ObjectExist(aws_key)) {
+            LOG(ERROR) << s3Info.bucketname() << " has been mounted, "
+                       << "use another bucket or delete the bucket objects!";
             return FSStatusCode::S3_INFO_ERROR;
+        } else {
+            if (s3Adapter_->PutObject(aws_key, "curvefs") != 0) {
+                LOG(ERROR) << "CreateFs " << fsName
+                       << " error, s3info is not available!";
+               return FSStatusCode::S3_INFO_ERROR;
+            }
```

Comment:
Can I understand this here, upload an s3 object named fsname to the bucket, and if the corresponding object exists, it is considered that fs already exists?
This doesn't feel very reasonable:
1. If fs is deleted, this object must be deleted together. Although there is a delete statement later, but the link is too long, if it fails in the middle, the object will remain.
2. If the cluster is dismantled directly. Then recreate it, obviously at this time the fs with the same name cannot be created
3. Considering the idempotency of creating fs, if multiple requests to create the same fs are sent at one time, only one will succeed.
It is recommended to upload an s3 object directly.

Reply:
1: If it fails in step one and two, three, the fs isn't deleted, it's ok. And If the fs status become Deleting in step 3, I will delete the object in the scanfs.
2: Do you mean two clusters will create the same fsname with the same bucket? 
3: multiple requests to create the same fs, I think all of it will succeed

Reply:
> 1: If it fails in step one and two, three, the fs isn't deleted, it's ok. And If the fs status become Deleting in step 3, I will delete the object in the scanfs. 2: Do you mean two clusters will create the same fsname with the same bucket? 3: multiple requests to create the same fs, I think all of it will succeed because of name lock

1. On the one hand, it is necessary to confirm that the intermediate creation fails to ensure that fs will not be deleted. In addition, considering backward compatibility, I think this kind of guarantee is unnecessary.

Reply:
> 1: If it fails in step one and two, three, the fs isn't deleted, it's ok. And If the fs status become Deleting in step 3, I will delete the object in the scanfs. 2: Do you mean two clusters will create the same fsname with the same bucket? 3: multiple requests to create the same fs, I think all of it will succeed because of name lock

1. On the one hand, it is necessary to confirm that the intermediate creation fails to ensure that fs will not be deleted. In addition, considering backward compatibility, I think this kind of guarantee is unnecessary.
2. In an extreme case, create a cluster first, then directly dismantle the cluster and create a new cluster when the s3 object is uploaded but not deleted. If the s3 is not cleaned up, create an fs with the same name in the new cluster. This situation should fail?
3. I think as long as you can upload an S3 object that fs will never use, this can be deleted or not.